### PR TITLE
[#232] Fix wrong groups pagination

### LINKED ISSUE #232: [BUG] Fix wrong groups pagination
**Describe the bug**
The main groups API has wrong pagination, because the private groups are removed as a second operation after executing the mongo query. As a result the groups count may be unexpectedly lower than the expected count 

**Fix**
The pagination and the private groups filtering are incorporated into the mongo query.

### PR DESCRIPTION
**Pull request by mdryankov**
## Description

**Resolves #232**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: driven/storage/adapter.go [hunk 1]
```diff
@@ -909,8 +909,40 @@ func (sa *Adapter) FindGroupByTitle(clientID string, title string) (*model.Group
...
+								},
+							},
+						},
+					},
+				},
+			},
+			bson.D{{"privacy", bson.D{{"$ne", "private"}}}},
+		}
+
+		if title != nil {
```
Comment (shurwit) [inline driven/storage/adapter.go:937]: Do we also need to account for the `hidden_for_search` field here?

File: core/administration.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ func (app *Application) getTODO() error {
 }
 
 func (app *Application) getGroupsUnprotected(clientID string, category *string, privacy *string, title *string, offset *int64, limit *int64, order *string) ([]model.Group, error) {
-	groups, err := app.storage.FindGroups(clientID, category, privacy, title, offset, limit, order)
+	groups, err := app.storage.FindGroups(clientID, nil, category, privacy, title, offset, limit, order)
```
Comment (shurwit) [inline core/administration.go:24]: Shouldn't something similar be used for the admin/v2/groups endpoint? Currently it is still filtering by user.

**Review by shurwit — COMMENTED** [review]
Thanks @mdryankov, this is looking good! I left a couple of comments below. 

Also if you wouldn't mind please take a look at the workflow described in https://github.com/rokwire/groups-building-block/blob/develop/CONTRIBUTING.md. It outlines some conventions that would be helpful for keeping things organized between teams, and for the benefit of clarity in the open source. Let me know if you have any questions or concerns about any of that. Thanks again!

File: driven/storage/adapter.go (hunk 1 above)
Comment (mdryankov) [reply driven/storage/adapter.go:937]: Nice catch! Thank you!

File: core/administration.go (hunk 2 above)
Comment (mdryankov) [reply core/administration.go:24]: The "administrator" should access everything (through the admin client) - this is the reason why not honouring the privacy, hidden flag and the members list.

**Pull request merged**